packages/wallets/src/features/cashier/modules/Transactions/components/TransactionsPendingRow/components/TransactionsPendingRowField/TransactionsPendingRowField.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useCallback, useRef } from 'react';
+import classNames from 'classnames';
+import { useHover } from 'usehooks-ts';
+import { Tooltip, WalletText } from '../../../../../../../../components/Base';
+import { useModal } from '../../../../../../../../components/ModalProvider';
+import useDevice from '../../../../../../../../hooks/useDevice';
+import { WalletActionModal } from '../../../../../../components/WalletActionModal';
+import './TransactionsPendingRowField.scss';
+
+type TProps = {
+    className?: classNames.ArgumentArray[number];
+    hint?: {
+        link: string;
+        text: string;
+        tooltipAlignment: React.ComponentProps<typeof Tooltip>['alignment'];
+    };
+    name: string;
+    value: string;
+    valueTextProps?: Omit<React.ComponentProps<typeof WalletText>, 'children'>;
+};
+
+const TransactionsPendingRowField: React.FC<TProps> = ({ className, hint, name, value, valueTextProps }) => {
+    const { isMobile } = useDevice();
+    const { show } = useModal();
+    const fieldRef = useRef(null);
+    const isFieldHovered = useHover(fieldRef);
+
+    const onValueClick = useCallback(() => {
+        if (hint)
+            isMobile
+                ? show(
+                      <WalletActionModal
+                          actionButtonsOptions={[
+                              {
+                                  isPrimary: true,
+                                  onClick: () => window.open(hint.link),
+                                  text: 'View',
+                              },
+                          ]}
+                          description={hint.text}
+                          title='Transaction details'
+                      />,
+                      { defaultRootId: 'wallets_modal_root' }
+                  )
+                : window.open(hint?.link);
+    }, [hint, isMobile, show]);
+
+    return (
+        <div className={classNames('wallets-transactions-pending-row-field', className)} key={name}>
+            <WalletText color='primary' size='xs'>
+                {name}
+            </WalletText>
+            {hint ? (
+                <Tooltip alignment={hint.tooltipAlignment} isVisible={!isMobile && isFieldHovered} message={hint.text}>
+                    <WalletText {...{ color: 'red', size: 'xs', weight: 'bold', ...valueTextProps }}>
+                        {isMobile ? (
+                            <button
```

Comment:
Why not use WalletButton here?

Reply:
This is a button only because of SonarCloud's errors 😅 It has all its button styles `unset`

Reply:
In other words, visually it's a `WalletText` but to have an `onClick` handler SonarCloud required we make it a button